pkg/apiserver/handlers.go

```diff
@@ -354,6 +355,18 @@ func NewRequestAttributeGetter(requestContextMapper api.RequestContextMapper, re
 	return &requestAttributeGetter{requestContextMapper, &APIRequestInfoResolver{util.NewStringSet(apiRoots...), restMapper}}
 }
 
+func (r *requestAttributeGetter) hasApiPrefix(req *http.Request) bool {
+	has := false
+	// Slice the first / of the path off (in apiRoots they're stored w/o leading /)
+	parts := strings.Split(req.URL.Path, "/")[1:]
+	// Paths with only one fragment won't have a prefix.
```

Comment:
This method name doesn't seem quite right.  It seems like the intent of this method is more `IsAPIResourceRequest`.  Otherwise, you could certainly have a request that is nothing but an APIPrefix.  This method returns false in that case.